R/data_tabulate.R

```diff
@@ -620,6 +627,34 @@ as.table.datawizard_crosstabs <- function(x, remove_na = TRUE, simplify = FALSE,
   isTRUE(attributes(x)$grouped_df)
 }
 
+.check_table_na <- function(x) {
+  # check if any table has NA values - the column "Value" contains the value
+  # "NA", and the column "N" contains the frequency of this value.
+  any(vapply(x, function(i) any(i$N[is.na(i$Value)] > 0), logical(1)))
+}
+
+.check_xtable_na <- function(x) {
+  any(vapply(x, function(i) {
+    browser()
+    # need to extract rownames, to check if we have a "NA" row
+    row_names <- as.character(i[[1]])
+    row_names[is.na(row_names)] <- "NA"
+    has_na <- FALSE
+    # check for NA columns and rows
+    if (!is.null(i[["NA"]])) {
+      has_na <- any(i[["NA"]] > 0)
+    }
+    if ("NA" %in% row_names) {
+      # we need "as.data.frame()" for grouped df
+      if (isTRUE(attributes(i)$grouped_df)) {
+        i$Group <- NULL
+      }
+      has_na <- has_na | any(as.vector(as.data.frame(i[row_names == "NA", -1])) > 0)
```

Comment:
Can `as.vector(as.data.frame())` be simplified to `as.vector()` only ? (I haven't tested)

Reply:
No, because for grouped_df, if a tidyverse package is loaded, simplification with `as.vector()` doesn't work, so we need a "real" data frame first.